include/sys/onoff.h

```diff
@@ -102,6 +108,29 @@ typedef void (*onoff_service_notify_fn)(struct onoff_service *srv,
 typedef void (*onoff_service_transition_fn)(struct onoff_service *srv,
 					    onoff_service_notify_fn notify);
 
+/** @brief On-off service transition functions. */
+struct onoff_service_transitions {
+	/* The function used to (initiate a) transition from off to on.  This
+	 * must not be null. Include @ref ONOFF_SERVICE_START_SLEEPS as
+	 * appropriate in flags.
+	 */
+	onoff_service_transition_fn start;
+
+	/* The function used to (initiate a) transition from on to off.  This
+	 * must not be null.  Include @ref ONOFF_SERVICE_STOP_SLEEPS as
+	 * appropriate in flags.
+	 */
+	onoff_service_transition_fn stop;
+
+	/* The function used to clear errors and force the service to an off
+	 * state.  Pass null if the service cannot or need not be reset.
+	 * (Services where a transition operation can complete with an error
+	 * notification should support the reset operation.) Include
+	 * @ref ONOFF_SERVICE_RESET_SLEEPS as appropriate in flags.
+	 */
+	onoff_service_transition_fn reset;
+};
```

Comment:
These should be documentation comments.  If they are multi-line, they should follow the implicit-brief style used for all other comments in this file, e.g.

```
	/**
	 * The function used to (initiate a) transition from off to on.
	 *
	 * This must not be null.
	 */
```

The information about `flags` doesn't really belong here if `flags` isn't part of this structure.  Thinking about it, though, the set of flags (restricted to the sleeps ones) *should* be stored here because it's tightly bound to the specific functions that are stored here.  Then the field here should be used to initialize the RAM copy in the service structure (which is in RAM because in principle I want services to be able to use the extra bits to record their own state).

So please add a `u16_t flags` field and update the initializer macros to take it.  Then the initializer macro for the service structure does not need to incorporate flags.

Hope that's clear.

Reply:
Agree that flags belong here but then it cannot be stored in ROM. Maybe flags should be split as they are combined into one variable mainly for optimization reasons.